pkg/policy/policy_controller.go

```diff
@@ -568,8 +571,61 @@ func (pc *PolicyController) updateUR(policy kyverno.PolicyInterface, urList []*u
 		return
 	}
 
+	pc.generateTargets(policy)
 	updateUR(pc.kyvernoClient, policy.GetName(), urList, pc.log.WithName("updateUR"))
+}
+
+// generateTargets ...
+func (pc *PolicyController) generateTargets(policy kyverno.PolicyInterface) {
```

Comment:
The error scenarios are not handled properly, we need to return errors and retry on failure.

Reply:
Once the UR created, IMO UpdateRequest controller should take care of reconcile if there are failures and update the correct status with failures info